kubecost/templates/cluster-controller/cluster-controller-deployment.yaml

```diff
@@ -174,8 +180,15 @@ spec:
           # and their validity.
           optional: true
       {{- if not (.Values.clusterController).legacyMode }}
+      {{- if and (not (.Values.clusterController).storageConfigSecret) (not (.Values.clusterController).storageConfig) }}
+      - name: federated-storage-config
+        secret:
+          defaultMode: 420
+          secretName: {{ include "kubecost.federatedStorage.secretName" . }}
+      {{- else }}
```

Comment:
We mount the federated storage Secret as a volume if neither of the cluster controller-specific `storageConfigSecret` and `storageConfig` are defined.